code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm

```diff
@@ -0,0 +1,48 @@
+/mob/living/carbon/superior_animal/roach/benzin
+	name = "Benzin Roach"
+	desc = "A monstrous, dog-sized cockroach. This one smells like welding fuel."
+	icon_state = "boomroach"
+	turns_per_move = 4
+	maxHealth = 25
+	health = 25
+	melee_damage_upper = 3
+	meat_type = /obj/item/reagent_containers/food/snacks/meat/roachmeat/benzin
+	meat_amount = 3
+	rarity_value = 15
+
+
+
+/mob/living/carbon/superior_animal/roach/benzin/Initialize(mapload)
+	. = ..()
+	set_light(0.5)
+	reagents.maximum_volume = 40
+
+
+/mob/living/carbon/superior_animal/roach/benzin/Life()
+	if(reagents.total_volume < reagents.maximum_volume)
+		reagents.add_reagent("fuel", 1)
+	..()
+
+
+/mob/living/carbon/superior_animal/roach/benzin/bullet_act(obj/item/projectile/P, def_zone)
+	. = ..()
+	if(prob(80))
+		explosion(get_turf(src), -1, -1, 2, 3)
+
+
+/mob/living/carbon/superior_animal/roach/benzin/attackby(obj/item/I, mob/living/user, params)
+	if(user.a_intent == I_HELP && istool(I))
+		var/obj/item/tool/T = I
+		if(T.use_fuel_cost)
```

Comment:
Does this actually let you refuel the tools? My brain isn't working right lately so I can't track this logic. If so, I love it.

Reply:
Holy shit it does.

Reply:
Y U P

Reply:
it also detonates when you melee them w/ a lit welder i'm fairly certain